atom/browser/lib/rpc-server.coffee

```diff
@@ -70,18 +74,30 @@ unwrapArgs = (sender, args) ->
         returnValue = metaToValue meta.value
         -> returnValue
       when 'function'
+        webContentsId = sender.getId()
+        rendererCallbacks = rendererRegistry[webContentsId]
+        if not rendererCallbacks?
+          # Weak reference to callbacks with their ID
+          rendererCallbacks = new IDWeakMap()
+          rendererRegistry[webContentsId] = rendererCallbacks
```

Comment:
The `rendererCallbacks` should be destroyed when the render view of webContents is destroyed.